code/modules/mob/living/living.dm

```diff
@@ -120,23 +120,7 @@ default behaviour is:
 			now_pushing = FALSE
 			return
 
-		if(istype(tmob, /mob/living/carbon/human) && (tmob.mutations & FAT))
-			if(prob(40) && !(mutations & FAT))
-				to_chat(src, SPAN_DANGER("You fail to push [tmob]'s fat ass out of the way."))
-				now_pushing = FALSE
-				return
-
-		if(istype(tmob.r_hand, /obj/item/shield/riot))
-			if(prob(99))
-				now_pushing = FALSE
-				return
-
-		if(istype(tmob.l_hand, /obj/item/shield/riot))
-			if(prob(99))
-				now_pushing = FALSE
-				return
-
-		if(!(tmob.status_flags & CANPUSH))
+		if(!(tmob.status_flags & CANPUSH) || (tmob.a_intent != I_HELP && HAS_TRAIT(tmob, TRAIT_UNPUSHABLE)))
```

Comment:
The trait should probably have limitations.

1. Direction - being unpushable from behind would be a bit weird.
2. Size - being unpushable by gigantic mobs would be very silly. A mech should still be able to move you, for instance, so should maybe a Ta or something really strong like a G2.